### ISSUE TITLE
[FEATURE] Disk Encryption Support

### ISSUE THREAD

**Issue by vtrenton** [harvester/harvester#1495]
**Is your feature request related to a problem? Please describe.**

With security being critical within datacenter infrastructure having encrypted volumes adds an additional layer of protection of data loss in the case of theft or pre-boot attacks.

**Describe the solution you'd like**

I would be nice to have the option to enable LUKS encryption of the root volume during installation.

**Comment by KyleSanderson** [comment]
Any update on this?

**Comment by MaxRenaud** [comment]
This is a blocker for us to use Harvester on some projects as well.

**Comment by innobead** [comment]
Duplicated. Let's track https://github.com/harvester/harvester/issues/3129 instead.

**Comment by tserong** [comment]
@innobead my read of this issue is that it's for encryption of the host disk(s), whereas #3129 is for encryption of VM volumes

**Comment by KyleSanderson** [comment]
@innobead boo.

**Comment by MaxRenaud** [comment]
@tserong this is also my understanding. Due to policy reasons, I am unable to install Harvester on a bare metal node without full-disk encryption. 

@innobead If this isn't something Harvester can/want to support, that's 100% fine. I want to make sure we're not closing something as a duplicate when it's not. 

Can you either reopen this issue or remove the duplicate tag and be explicit that full-disk encryption on bare metal isn't considered?

**Comment by innobead** [comment]
Thanks for clarifying the context, as this was mistakenly regarded as the volume encryption feature.

**Comment by johanot** [comment]
It's somewhat straight forward to add a an optional LUKS envelope at install time. However, any thoughts on which key providers to support at boot time then? Anyone?

The initial implementation could be just "interactive passphrase on the console". But... In that case, it doesn't become usable for me before I _also_ have network+ssh in initrd - since Hetzner don't provide console access by default.

**Comment by MaxRenaud** [comment]
For us, we'd be looking at having a hardware TPM handle key managment otherwise it doesn't scale.

**Comment by bcdurden** [comment]
Add my customers as ones that like the TPM key management for the same reasons.

**Comment by cromulentbanana** [comment]
> The initial implementation could be just "interactive passphrase on the console". But... In that case, it doesn't become usable for me before I _also_ have network+ssh in initrd - since Hetzner don't provide console access by default.

at a minimum, network+ssh in initrd

**Comment by andrew-morris-rgs** [comment]
Would adding a LUKS device by default and pointing it to just a keyfile to start be an acceptable catchall. With TPM, network, and password decryption all being optional replacements of that after the fact?

The change would have to go here right? Or is there an other place more suited?
https://github.com/harvester/harvester-installer/blob/34a844549a55248e9db02022487405b7268f7522/package/harvester-os/files/usr/sbin/harv-install

**Comment by innobead** [comment]
This is blocked by https://github.com/harvester/harvester/issues/10555. Moved to planning first until that issue is fixed. 

cc @harvester/pm @tserong @harvester/tpm
